### ISSUE TITLE
Adding a jsx comment causes TypeScript to identify children type as any[], even though generated javascript doesn't change

### ISSUE THREAD

**Issue by lxe** [microsoft/TypeScript#41389]
**TypeScript Version:**  3.3.3 - 4.0.5 

**Search Terms:**

Typescript jsx comment any[] children

**Code**

[Playground](https://www.typescriptlang.org/play?ts=4.0.5#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcA5FDvmQNwBQdAJlrgDbKVwBuHcAyiSwBhEpAB2WMTABcmKjAB0AMSEAeAN5042uLgAWwVo0piA-LOx5Fl-AFFWWEJJh0AvgD56dAgFcx+YAgxOABxCAhGOAAKAEo4TR04AHokuABNYCwjdBtFXEpkGCx7R2cogScRcCDnABo4MR9WVnrchXycIpKnKSiAIkZgLj76xuaYifpElLgAFQN0VELgVAJM9FmATzAsPnzgMBdEyhgfKGDVCuFRGql3LUSdVUGuZPvE1SSrqvFne9cGL5-DBAsEAELISKxeIPbQzPgg5pwTaZbJyKztApdBw9GDlQQ-W4wUZNFro-CYzrFHFlAZDEYNUkTGJTHQzeYrOCMCBYdAAOQA8rM4EsQatNnNtrt9oc4DApbo9CwANbAMQAc2ksOSqS2O3I6kVhmMklkyDEmwA2gBdGhwVxkOCcsQQeBoVDAdViZAAIwccogcoVZEN+mNJnM5Jg3WcqgU8bICm1JzOF2+Nwkd21iXUSSQVVxiCSAMeTxeb21n3T1UzMH+DDoQA)

```ts
import * as React from 'react';

declare var SomeComponent: React.FC<{
    children?: React.ReactElement
}>;

function Good () {
    // Yields React.createElement(SomeComponent, null, React.createElement("div", null)));
    // This satisfies TypeScript
    return <SomeComponent>
        <div />
    </SomeComponent>
}

function Bad () {
    // Still yields React.createElement(SomeComponent, null, React.createElement("div", null)));
    // This does NOT satisfy TypeScript type checking:
    // Type '{ children: any[]; }' is not assignable to type '{ children?: ReactElement<...'.
    return <SomeComponent>
        {/* Comment */}
        <div />
    </SomeComponent>
}
```

**Expected behavior:**

The case without a JSX comment and the case with a comment both yield the same JavaScript (e.g. the comment does not count as an extra child). TypeScript should understand that the type of `<div />` matches the expected signature `React.ReactElement`

**Actual behavior:**

However, when the JSX comment is present, even though the resulting JS is the same, TypeScript now assumes the type to be `any[]` which no longer satisfies `React.ReactElement`.

**Playground Link:** [Link](https://www.typescriptlang.org/play?ts=4.0.5#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcA5FDvmQNwBQdAJlrgDbKVwBuHcAyiSwBhEpAB2WMTABcmKjAB0AMSEAeAN5042uLgAWwVo0piA-LOx5Fl-AFFWWEJJh0AvgD56dAgFcx+YAgxOABxCAhGOAAKAEo4TR04AHokuABNYCwjdBtFXEpkGCx7R2cogScRcCDnABo4MR9WVnrchXycIpKnKSiAIkZgLj76xuaYifpElLgAFQN0VELgVAJM9FmATzAsPnzgMBdEyhgfKGDVCuFRGql3LUSdVUGuZPvE1SSrqvFne9cGL5-DBAsEAELISKxeIPbQzPgg5pwTaZbJyKztApdBw9GDlQQ-W4wUZNFro-CYzrFHFlAZDEYNUkTGJTHQzeYrOCMCBYdAAOQA8rM4EsQatNnNtrt9oc4DApbo9CwANbAMQAc2ksOSqS2O3I6kVhmMklkyDEmwA2gBdGhwVxkOCcsQQeBoVDAdViZAAIwccogcoVZEN+mNJnM5Jg3WcqgU8bICm1JzOF2+Nwkd21iXUSSQVVxiCSAMeTxeb21n3T1UzMH+DDoQA)

**Related Issues:** 

 - https://stackoverflow.com/questions/64667775/is-it-possible-to-prevent-jsx-comment-from-adding-a-children-element-potentia
 - https://github.com/microsoft/TypeScript/issues/27552#issuecomment-719829777

**Comment by MartinJohns** [comment]
Very likely a duplicate of #41125.

**Comment by lxe** [comment]
Thanks @MartinJohns that's gotta be it! No issues in nightly! https://www.typescriptlang.org/play?ts=4.2.0-dev.20201103#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcA5FDvmQNwBQdAJlrgDbKVwBuHcAyiSwBhEpAB2WMTABcmKjAB0AMSEAeAN5042uLgAWwVo0piA-LOx5Fl-AFFWWEJJh0AvgD56dAgFcx+YAgxOABxCAhGOAAKAEo4TR04AHokuABNYCwjdBtFXEpkGCx7R2cogScRcCDnABo4MR9WVnrchXycIpKnKSiAIkZgLj76xuaYifpElLgAFQN0VELgVAJM9FmATzAsPnzgMBdEyhgfKGDVCuFRGql3LUSdVUGuZPvE1SSrqvFne9cGL5-DBAsEAELISKxeIPbQzPgg5pwTaZbJyKztApdBw9GDlQQ-W4wUZNFro-CYzrFHFlAZDEYNUkTGJTHQzeYrOCMCBYdAAOQA8rM4EsQatNnNtrt9oc4DApbo9CwANbAMQAc2ksOSqS2O3I6kVhmMklkyDEmwA2gBdGhwVxkOCcsQQeBoVDAdViZAAIwccogcoVZEN+mNJnM5Jg3WcqgU8bICm1JzOF2+Nwkd21iXUSSQVVxiCSAMeTxeb21n3T1UzMH+DDoQA

**Comment by RyanCavanaugh** [comment]
Things merged prior to last week will be in 4.1. Thanks!
